src/Interpreters/ProcessList.cpp

```diff
@@ -430,8 +438,6 @@ QueryStatus * ProcessList::tryGetProcessListElement(const String & current_query
 
 CancellationCode ProcessList::sendCancelToQuery(const String & current_query_id, const String & current_user, bool kill)
 {
-    auto lock = safeLock();
-
     QueryStatus * elem = tryGetProcessListElement(current_query_id, current_user);
```

Comment:
Seems like race condition is possible now: if query finishes just after we got raw pointer to `QueryStatus` then this pointer may be invalid